### PR TITLE
new issue templates

### PR DESCRIPTION
**Pull request by jmklix**
*Issue #, if available:*

*Description of changes:*
Use new issue templates and direct new guidance questions to discussions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1 +1,6 @@
+---
 blank_issues_enabled: false
+contact_links:
+  - name: 💬 General Question
+    url: https://github.com/aws/aws-iot-device-sdk-java-v2/discussions/categories/q-a
```
Comment (jdisanti) [inline .github/ISSUE_TEMPLATE/config.yml:5]: Is this URL wrong?

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-      description: |
-        List the versions & crates of the aws-rust-sdk you are using.
-
-        `cargo install cargo-tree`
-        (see install here: https://github.com/sfackler/cargo-tree)
-
-        Then:
-        `cargo tree | grep aws-`
-    validations:
-      required: true
```
Comment (jdisanti) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:26]: This needs to be kept.

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 3]
```diff
@@ -1,78 +0,0 @@
...
-        if you haven't already (e.g., `tracing_subscriber::fmt::init();`),
-        and then setting the environment variable `RUST_LOG` before
-        running your program, as follows:
-        `RUST_LOG='aws_smithy_http_tower::dispatch=trace,aws_smithy_http::middleware=trace'`
-
-        For example:
-        `RUST_LOG='aws_smithy_http_tower::dispatch=trace,aws_smithy_http::middleware=trace' cargo run`
-
-        The SDK redacts sensitive information such as auth headers in these trace logs,
-        but please look through them before posting just to be sure.
```
Comment (jdisanti) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:78]: This needs to be kept.

File: .github/ISSUE_TEMPLATE/---feature-request.yml [hunk 4]
```diff
@@ -1,50 +0,0 @@
...
-  - type: textarea
-    attributes:
-      label: A note for the community
-      value: |
-        <!-- Please keep this note for the community -->
-        ### Community Note
-        * Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
-        * Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
-        * If you are interested in working on this issue, please leave a comment
-        <!-- Thank you for keeping this note for the community -->
```
Comment (jdisanti) [inline .github/ISSUE_TEMPLATE/---feature-request.yml:15]: This needs to be kept.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (jmklix) [reply .github/ISSUE_TEMPLATE/config.yml]: fixed

**Review by jdisanti — APPROVED** [review]

**Pull request merged**